src/tests/end2end/D3D12ResidencyManagerTests.cpp

```diff
@@ -311,23 +320,23 @@ TEST_F(D3D12ResidencyManagerTests, OverBudgetWithGrowth) {
     std::vector<ComPtr<ResourceAllocation>> allocations = {};
     std::vector<Heap*> resourceHeaps = {};
 
-    const D3D12_RESOURCE_DESC bufferDesc = CreateBasicBufferDesc(GPGMM_MB_TO_BYTES(1));
+    constexpr uint64_t kBufferMemorySize = GPGMM_MB_TO_BYTES(1);
+    const D3D12_RESOURCE_DESC bufferDesc = CreateBasicBufferDesc(kBufferMemorySize);
 
-    while (!IsOverBudget(residencyManager.Get())) {
+    ALLOCATION_DESC bufferAllocationDesc = {};
+    bufferAllocationDesc.HeapType = D3D12_HEAP_TYPE_DEFAULT;
+
+    const DXGI_MEMORY_SEGMENT_GROUP bufferMemorySegment =
+        residencyManager->GetMemorySegmentGroup(bufferAllocationDesc.HeapType);
+    const uint64_t memoryUnderBudget = GetBudgetLeft(residencyManager.Get(), bufferMemorySegment);
+
+    while (resourceAllocator->GetInfo().UsedMemoryUsage + kBufferMemorySize < memoryUnderBudget) {
         ComPtr<ResourceAllocation> allocation;
         ASSERT_SUCCEEDED(resourceAllocator->CreateResource(
-            {}, bufferDesc, D3D12_RESOURCE_STATE_COMMON, nullptr, &allocation));
+            bufferAllocationDesc, bufferDesc, D3D12_RESOURCE_STATE_COMMON, nullptr, &allocation));
 
         resourceHeaps.push_back(allocation->GetMemory());
         allocations.push_back(std::move(allocation));
-
-        // Prevent the first created resources from being evicted once over budget.
-        ASSERT_SUCCEEDED(residencyManager->UpdateVideoMemorySegments());
-    }
-
-    // Created allocations above the budget should be resident.
-    for (auto& allocation : allocations) {
```

Comment:
Note: this was removed because we can't guarentee kBufferMemorySize is always the same due to pre-allocation being enabled.